pwa/app/lib/matchUtils.ts

```diff
@@ -67,3 +67,34 @@ export function matchTitleShort(match: Match, event: Event): string {
   console.log(match.key, event);
   return 'IF YOU SEE THIS PLEASE PING JUSTIN/EUGENE WITH EVENT KEY';
 }
+
+export function calculateTeamRecordFromMatches(
+  teamKey: string,
+  matches: Match[],
+): WltRecord {
+  const won = matches.filter(
+    (m) =>
+      (m.winning_alliance === 'red' &&
+        m.alliances.red.team_keys.includes(teamKey)) ||
+      (m.winning_alliance === 'blue' &&
+        m.alliances.blue.team_keys.includes(teamKey)),
+  ).length;
+
+  const lost = matches.filter(
+    (m) =>
+      (m.winning_alliance === 'red' &&
+        m.alliances.blue.team_keys.includes(teamKey)) ||
+      (m.winning_alliance === 'blue' &&
+        m.alliances.red.team_keys.includes(teamKey)),
+  ).length;
+
+  const tied = matches.filter(
+    (m) =>
+      (m.alliances.blue.team_keys.includes(teamKey) ||
+        m.alliances.red.team_keys.includes(teamKey)) &&
+      m.winning_alliance === '' &&
+      m.actual_time !== null,
```

Comment:
better to check that score is not `-1` to see if a match has been played. Not all matches have `actual_time` populated.

I'd create a helper called `hasBeenPlayed(match)` that checks if both red and blue scores are not `-1`.